addons/stock/models/stock_location.py

```diff
@@ -167,6 +167,12 @@ def _onchange_usage(self):
         if self.usage not in ('internal', 'inventory'):
             self.scrap_location = False
 
+    @api.constrains('scrap_location')
+    def _check_scrap_location(self):
+        for record in self:
+            if record.scrap_location and self.env['stock.picking.type'].search_count([('code', '=', 'mrp_operation'), '|', ('default_location_src_id', '=', record.id), ('default_location_dest_id', '=', record.id)]):
```

Comment:
Do we also need to restrict the source location too? I would think sometimes you could want to reuse scrap components. It also doesn't throw a traceback in this case as well (same comment applies to the `_check_default_location`)

Side note, you can use `search` instead of `search_count` since an empty recordset is also Falsy, but it doesn't really matter in this case.